examples/llama_1.1b/ex_config_quantization.py

```diff
@@ -12,76 +12,120 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from tqdm import tqdm
-from torch.utils.data import RandomSampler
+####
+#
+# The following example shows how a model can be calibrated and
+# compressed entirely with primitives within `compressed-tensors`
+# using PyTorch hooks.
+# The resulting model's .safetensors file should be 1.2GB,
+# whereas the original model's .safetensors file is 4.1GB.
+# See `./ex_llmcompressor_quantization.py` for how this can be
+# simplified using the vllm's `llm-compressor` package
+#
+####
+
+from pathlib import Path
+
+import torch
+from compressed_tensors.compressors import ModelCompressor
 from compressed_tensors.quantization import (
-    apply_quantization_config,
-    freeze_module_quantization,
     QuantizationConfig,
     QuantizationStatus,
+    apply_quantization_config,
 )
-from sparseml.transformers.finetune.data.data_args import DataTrainingArguments
-from sparseml.transformers.finetune.data.base import TextGenerationDataset
+from datasets import load_dataset
+from torch.utils.data import DataLoader, RandomSampler
+from tqdm import tqdm
 from transformers import AutoModelForCausalLM, AutoTokenizer, DefaultDataCollator
-from torch.utils.data import DataLoader
-from sparseml.pytorch.utils import tensors_to_device
-import torch
 
-config_file = "example_quant_config.json"
+
+config_file = Path(__file__).parent / "example_quant_config.json"
 model_name = "TinyLlama/TinyLlama-1.1B-intermediate-step-1431k-3T"
-dataset_name = "open_platypus"
+dataset_name = "garage-bAInd/Open-Platypus"
 split = "train"
 num_calibration_samples = 512
 max_seq_length = 1024
 pad_to_max_length = False
 output_dir = "./llama1.1b_new_quant_out"
 device = "cuda:0" if torch.cuda.is_available() else "cpu"
 
-model = AutoModelForCausalLM.from_pretrained(model_name, device_map=device, torch_dtype="auto")
+model = AutoModelForCausalLM.from_pretrained(
+    model_name, device_map=device, torch_dtype="auto"
+)
 model.eval()  # no grad or updates needed for base model
-config = QuantizationConfig.parse_file(config_file)
+config = QuantizationConfig.model_validate_json(config_file.read_text())
 
 # set status to calibration
 config.quantization_status = QuantizationStatus.CALIBRATION
 
 # initialize quantization
 apply_quantization_config(model, config)
 
+
+# create hook to keep track of scales and zero points on each module with a quantization_scheme
```

Comment:
would update this to be:
"to update scales and zero points"

Reply:
sounds good, updated!